README.md

```diff
@@ -13,18 +13,39 @@ element's [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Componen
 and renders declaratively using `lit-html`.
 
-  * **Setup properties:** LitElement supports a static `properties` getter in which element property
-  accessors are described. For each property, an object configures settings where the options are:
-
-    * attribute: if false, do not add to observedAttributes, if true or absent, observe the
-    lowercased property name, if a string observe that value.
-    * type: if a function, use to deserialize the attribute value to the property value;
-    also can be an object with `{fromAttribute, toAttribute}` where `fromAttribute` is
-    the deserialize function and `toAttribute` is a serialize function.
-    * reflect: if true, on setting the property reflects the property value to an
-    attribute value using `type.toAttribute` if it exists.
-    * shouldInvalidate: optional function which should return true if setting the property
-    should cause the element to invalidate causing it to asynchronously update.
+  * **Setup properties:** LitElement supports observable properties which may trigger an
+  update when set. These properties can be written in a few ways:
+
+    * As class fields with the `@property()` [decorator](https://github.com/tc39/proposal-decorators#decorators),
+    if you're using a compiler that supports them, like TypeScript or Babel.
+    * With a static `properties` getter.
+    * By manually writing getters and setters. You can call `setProperty` or `invalidate`
+    to trigger an update. If you use `setProperty`, you can use `getProperty` in the getter.
+
+    Properties may be given an options argument which is an object that desribes how to
+    process the property. This may can be either in the `@property({...})` decorator or in the
+    object returned the `properties` getter, e.g. `static get properties { return { foo: {...} }`.
+
+    Property options include:
+
+    * `attribute`: Describes how and if the property becomes an observedAttribute.
+    If the value is false, the property is not added to `observedAttributes`.
+    If true or absent, the lowercased property name is observed (e.g. `fooBar` becomes `fobar`).
+    If a string, the string value is observed (e.g `attribute: 'foo-bar'`).
+    * `type`: Describes how to convert the attribute to/from a property.
+    If this value is a function, it is used to deserialize the attribute value
+    a the property value. If it's an object, it can have keys for `fromAttribute` and
+    `toAttribute` where `fromAttribute` is the deserialize function and `toAttribute`
+    is a serialize function used to convert the property to an attribute.
+    * `reflect`: Describes if the property should reflect to an attribute.
+    If true, when the property is set, the attribute value is set using the
+    attribute name taken from the property's `attribute` and the value
+    of the property serialized using `type.toAttribute` if it exists.
+    * `shouldInvalidate`: Describes if setting a property should trigger
```

Comment:
I think this name might be a bit confusing, because as a dev I would think "why shouldn't it invalidade?"

But when I read here it is more understandable that for complex objects equality is much more complex. Maybe I would turn it around and call it `isEqual` 

> `isEqual`: An equality comparisons function used for checking whether the property change does not need re-render. If not present, a strict identity check (=== operator) is used. This function takes the arguments `newValue` and `oldValue` and return `true` if the two values are considered equals, and thus no re-render is needed. This is useful if a property should be considered dirty only if some condition is met, like if a key of an object value changes.

Reply:
Going to leave it as is for now to get some more feedback but will consider changing it.